script/deploy/mainnet/M2_Mainnet_Upgrade.s.sol

```diff
@@ -128,3 +130,179 @@ contract M2_Mainnet_Upgrade is ExistingDeploymentParser {
+
+        Tx[] memory txs = new Tx[](11);
+        txs[0] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(delegationManager))), 
+                delegationManagerImplementation
+            )
+        );
+
+        txs[1] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(slasher))), 
+                slasherImplementation
+            )
+        );
+
+        txs[2] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(strategyManager))), 
+                strategyManagerImplementation
+            )
+        );
+
+        txs[3] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(delayedWithdrawalRouter))), 
+                delayedWithdrawalRouterImplementation
+            )
+        );
+
+        txs[4] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(eigenPodManager))), 
+                eigenPodManagerImplementation
+            )
+        );
+
+        txs[5] = Tx(
+            address(eigenPodBeacon),
+            0,
+            abi.encodeWithSelector(
+                UpgradeableBeacon.upgradeTo.selector, 
+                eigenPodImplementation
+            )
+        );
+
+        // unpause everything on DelegationManager
+        txs[6] = Tx(
+            address(delegationManager), 
+            0, // value
+            abi.encodeWithSelector(Pausable.unpause.selector, 0)
+        );
+
+        // set the min withdrawal delay blocks on the DelegationManager
+        txs[7] = Tx(
+            address(delegationManager), 
+            0, // value
+            abi.encodeWithSelector(DelegationManager.setMinWithdrawalDelayBlocks.selector, DELEGATION_MANAGER_MIN_WITHDRAWAL_DELAY_BLOCKS)
+        );
+
+        // unpause everything on EigenPodManager
+        txs[8] = Tx(
+            address(eigenPodManager), 
+            0, // value
+            abi.encodeWithSelector(Pausable.unpause.selector, 0)
+        );
+
+        // set beacon chain oracle on EigenPodManager
+        txs[9] = Tx(
+            address(eigenPodManager), 
+            0, // value
+            abi.encodeWithSelector(EigenPodManager.updateBeaconChainOracle.selector, IBeaconChainOracle(beaconChainOracle))
```

Comment:
Can just use `beaconOracle` here instead of  `IBeaconChainOracle(beaconChainOracle)`. It will be the same address as the one read from "script/configs/mainnet/M2_mainnet_upgrade.config.json"

Reply:
Guess not a big deal to see actual address we're using here